lib/beaglebone.js

```diff
@@ -365,7 +369,195 @@ BeagleBone.prototype.sendI2CReadRequest = function(address, length, cb) {
+    return this;
+  }
+
+  // Fix arguments if called with Firmata.js API
+  if (arguments.length === 2) {
+    if (Array.isArray(cmdRegOrData)) {
+      inBytes = cmdRegOrData.slice();
+      cmdRegOrData = inBytes.shift();
+    } else {
+      inBytes = [];
+    }
+  }
+
+  buffer = new Buffer([cmdRegOrData].concat(inBytes));
+
+  // Only write if bytes provided
+  if (buffer.length) {
+    _i2c.i2cWriteSync(address, buffer.length, buffer);
+  }
+  return this;
+};
+
+BeagleBone.prototype.i2cWriteReg = function(address, register, value) {
+  if (_i2c === undefined) {
+    this.i2cConfig();
+  }
+  _i2c.writeByteSync(address, register, value);
+  return this;
+};
+
+// this method supports both
+// i2cRead(address, register, bytesToRead, handler)
+// and
+// i2cRead(address, bytesToRead, handler)
+BeagleBone.prototype.i2cRead = function(address, register, bytesToRead, handler) {
+
+  var event = "I2C-reply" + address + "-";
+  var buffer = new Buffer(bytesToRead);
+
+  if (_i2c === null) {
+    this.i2cConfig();
+  }
+
+  // Fix arguments if called with Firmata.js API
+  if (arguments.length === 3 &&
+    typeof register === "number" &&
+    typeof bytesToRead === "function") {
+    handler = bytesToRead;
+    bytesToRead = register;
+    register = null;
+  }
+
+  handler = typeof handler === "function" ? handler : function() {};
+
+  if (register !== null) {
+    this.i2cWrite(address, register);
+  } else {
+    register = 0;
+  }
+
+  event += register;
+
+  this.on(event, readLoop);
+
+  function readLoop() {
+    _i2c.i2cRead(address, bytesToRead, buffer, function(i2cError, i2cNumBytes, buffer) {
+      if (i2cError) {
+        this.emit("error", i2cError);
+        return;
+      }
+      this.emit(event, buffer);
+      handler(buffer);
+    });
+    setTimeout(readLoop, _i2cDelay);
+  }
 
+  return this;
+};
+
+// this method supports both
+// i2cReadOnce(address, register, bytesToRead, handler)
+// and
+// i2cReadOnce(address, bytesToRead, handler)
+BeagleBone.prototype.i2cReadOnce = function(address, register, bytesToRead, handler) {
+  var event = "I2C-reply" + address + "-";
+  var buffer = new Buffer(bytesToRead);
+
+  if (_i2c === null) {
+    this.i2cConfig();
+  }
+
+  // Fix arguments if called with Firmata.js API
+  if (arguments.length === 3 &&
+    typeof register === "number" &&
+    typeof bytesToRead === "function") {
+    handler = bytesToRead;
+    bytesToRead = register;
+    register = null;
+  }
+
+  handler = typeof handler === "function" ? handler : function() {};
+
+  if (register !== null) {
+    this.i2cWrite(address, register);
+  } else {
+    register = 0;
+  }
+
+  event += register;
+
+  this.once(event, handler);
+
+  function readLoop() {
+    _i2c.i2cRead(address, bytesToRead, buffer, function(i2cError, i2cNumBytes, buffer) {
+      if (i2cError) {
+        this.emit("error", i2cError);
+        return;
+      }
+      this.emit(event, buffer);
```

Comment:
`this` is not bound to the correct object here so calling emit will result in an exception being thrown.